unstable/wlr-output-manager-v1.xml

```diff
@@ -0,0 +1,350 @@
+
+        This event is sent after the client binds to the output_device and
+        whenever the mode of the output changes.
+      </description>
+      <arg name="transformation" type="int" enum="wl_output.transform"/>
+    </event>
+    
+    <event name="size">
+      <description summary="size of the output in the compositor space">
+        The size event describes the size of the output in the global
+        compositor space.
+
+        It is comparable to the zxdg_output_v1.logical_size event, but is not
+        the same in all circumstances. The zxdg_output_v1.logical_size event
+        will not be the same if the compositor does not scale the surface
+        buffers.
+
+        Combined with the current_mode and transformation event, this event
+        effectively describes the scale applied to the output's buffer
+        before it is displayed.
+
+        This event is sent after the client binds to the output_device object
+        and whenever the size of the output device in the global compositor
+        space changes.
+      </description>
+      <arg name="width" type="int"
+        summary="width in global compositor space"/>
+      <arg name="height" type="int"
+        summary="height in global compositor space"/>
+    </event>
+
+    <event name="position">
+      <description summary="position of the output in the compositor space">
+        This events describes the position of the output_device within the
+        global compositor space.
+
+        If the compositor implements xdg_output and this output is enabled,
+        the zxdg_output_v1.logical_position must report the same position. 
+
+        This event is sent after the client binds to the output_device
+        and whenever the position of the output changes in the compositor space.
+      </description>
+      <arg name="x" type="int"
+        summary="x position within the global compositor space"/>
+      <arg name="y" type="int"
+        summary="y position within the global compositor space"/>
+    </event>
+    
+    <enum name="enablement">
+      <description summary="describes enabled state">
+        Describes whether a device is enabled, i.e. device is used to
+        display content by the compositor. This wraps a boolean around
+        an int to avoid a boolean trap.
+      </description>
+      <entry name="disabled" value="0"/>
+      <entry name="enabled" value="1"/>
+    </enum>
+
+    <event name="enabled">
+      <description summary="output is enabled or disabled">
+        This event describes whether the output is actively displaying a part
+        of the global compositor space.
+
+        If an output_device is enabled, the output must be represented by
+        a wl_output.
+        
+        This event is sent after the client binds to the output_device
+        and whenever the output device is enabled or disabled.
+      </description>
+      <arg name="enabled" type="int" summary="output enabled state"/>
```

Comment:
Instead of having an enum, we can just have an `int` which is non-zero when enabled

Reply:
Or something like:
`
<event name="wl_output>
<arg name="output" type="object" interface="wl_output" nullable/>
<event>
`

non-null when it's active, and null when sent to inactivity.
A direct link to wl_output (avoiding string comparisons) is missing either way IMO

Reply:
I'm not against a link between this and wl_output, but iirc there was some discussion on #wayland about different protocols creating the same interfaces and why that was a bad idea. Might be wrong on that. This would also be a server-created object, which is tricky to destroy. (see http://ppaalanen.blogspot.com/2014/07/wayland-protocol-design-object-lifespan.html )

Reply:
"A link to wl_output" would not mean that this interface would create wl_output objects.

However it is tricky, because output_device objects are published as globals, and we can "link" wl_outputs only after the client has bound to the global…

Reply:
Ah, got it, that would be tricky indeed.

Reply:
The more I think about this, the more I don't like this approach. An alternative solution would be  to send the name that the registry gives to the global. However, as far as I can tell, it's not possible to obtain that in the current api.

Reply:
Should be client specific, because you may hide others, and they are just a counter from what I've seen.

Well, it would be sent with all other events on binding. It can't be linked to it like xdg-output is either way. It's just an "attribute" the client can track.

Reply:
Looking at the libwayland code, the names are indeed a counter. However, there's currently no way to obtain the name of a global, except guessing it.